prometheus/testutil/difflib.go

```diff
@@ -0,0 +1,649 @@
+// Copyright 2022 The Prometheus Authors
```

Comment:
I believe we want to make sure this is under `internal` path so no one can import it. Otherwise we have to care about API, Go doc comments etc. 

And I don't client_golang have to be library which offers diffing 🤔